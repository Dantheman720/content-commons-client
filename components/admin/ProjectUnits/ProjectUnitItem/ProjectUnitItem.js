import React, {
  useState, useEffect, useContext
} from 'react';
import dynamic from 'next/dynamic';
import PropTypes from 'prop-types';
import { getCount, getPathToS3Bucket } from 'lib/utils';
import truncate from 'lodash/truncate';
import {
  Card, Modal, Image, List, Loader
} from 'semantic-ui-react';
import iconVideoCamera from 'static/icons/icon_32px_videoCamera.png';
import FileUploadProgressBar from 'components/admin/ProjectEdit/FileUploadProgressBar/FileUploadProgressBar';
import GeneralError from 'components/errors/GeneralError/GeneralError';
import { UploadContext } from 'components/admin/ProjectEdit/VideoEdit/VideoEdit';
import './ProjectUnitItem.scss';

const EditSingleProjectItem = dynamic( () => import( /* webpackChunkName: "editSingleProjectItem" */ 'components/admin/ProjectEdit/EditSingleProjectItem/EditSingleProjectItem' ) );

const ProjectUnitItem = props => {
  const { projectId, unit, filesToUpload } = props;
  const PLACEHOLDER = null;
  const [thumbnail, setThumbnail] = useState( PLACEHOLDER );
  const [title, setTitle] = useState( '' );
  const [unitUploadComplete, setUnitUploadComplete] = useState( false );
  const [error, setError] = useState( false );

  // do not put in state as it will create a new object and will not track upload
  // need a better solution as this is prone to errors
  const unitFileToUpload = filesToUpload.filter( file => file.language === unit.language.id );

  const uploadInProgress = useContext( UploadContext );

  // implemment subscriptions to track thumbnail changes
  const getThumbnail = u => {
    if ( u && u.thumbnails && u.thumbnails[0] && u.thumbnails[0].image ) {
      return `${getPathToS3Bucket()}/${u.thumbnails[0].image.url}`;
    }
    // return PLACEHOLDER;
  };

  const getFileStream = ( file = {}, site = 'vimeo' ) => {
    if ( file.stream && Array.isArray( file.stream ) ) {
      return file.stream.find( stream => stream.site === site );
    }
    return {};
  };

  const hasError = () => {
    if ( unit && Array.isArray( unit.files ) ) {
      return unit.files.some( file => {
        const stream = getFileStream( file );
        if ( file.error ) {
          return file.error;
        }
        return getCount( stream ) && !stream.url;
      } );
    }
    return false;
  };

  const handleOnComplete = () => {
    setUnitUploadComplete( true );
  };

  useEffect( () => {
    setThumbnail( getThumbnail( unit ) );
    setTitle( unit && unit.title ? unit.title : '[Title]' );
  }, [unit] );

  useEffect( () => {
    setError( hasError() );
  }, [projectId] );

  const renderProjectItem = () => (
    <Card className="project-unit-item">
      <div className="image-wrapper">
        { thumbnail
          ? ( <Image src={ thumbnail } fluid /> )
          : <div className="placeholder" />
        }
        { /* Has an upload been triggered and if so, are the files applicable to this unit complete? */ }
        { uploadInProgress && !unitUploadComplete && ( <Loader active size="small" /> ) }

        { /* Remove file overlay if above conditions are met */ }
        { !!unitFileToUpload.length && !unitUploadComplete && (
          <List className="file-list">
            { unitFileToUpload.map( file => <List.Item key={ file.id }>{ truncate( file.input.name, { length: 45 } ) }</List.Item> ) }
          </List>
        )
        }

        { !error
          && <img className="metaicon" src={ iconVideoCamera } alt="Video Icon" /> }
      </div>

      { /* Remove progress bar if above conditions are met */ }
      { uploadInProgress && !unitUploadComplete && !!unitFileToUpload.length && (
        <FileUploadProgressBar
          filesToUpload={ unitFileToUpload }
          label="Upload in progress"
          labelAlign="left"
          barSize="tiny"
          onComplete={ handleOnComplete }
          showPercent
        />
      ) }
      <Card.Content className={ error ? 'error' : '' }>
        <Card.Header>{ title }</Card.Header>
        <Card.Meta>
          <List>
            <List.Item>Language: { unit.language.displayName }</List.Item>
            <List.Item>Files: { unit.files.length }</List.Item>
            { error
              && (
                <List.Item>
                  <GeneralError msg="Uploading Error" />
                </List.Item>
              ) }
          </List>
        </Card.Meta>
      </Card.Content>
    </Card>
  );

  if ( projectId && !error ) {
    return (
      <Modal
        key={ unit.language.id }
        trigger={ renderProjectItem() }
        closeIcon
      >
        <Modal.Content>
          <EditSingleProjectItem
            projectId={ projectId }
            itemId={ unit.id }
          />
        </Modal.Content>
      </Modal>
    );
  }

  // this is a new project, deactivate modal
  return renderProjectItem();
};

ProjectUnitItem.propTypes = {
  unit: PropTypes.object,
  filesToUpload: PropTypes.array,
  projectId: PropTypes.string
};


export default ProjectUnitItem;
