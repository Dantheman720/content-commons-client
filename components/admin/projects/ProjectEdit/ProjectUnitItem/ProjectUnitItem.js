import React, {
  useState, useEffect, useContext
} from 'react';
import dynamic from 'next/dynamic';
import PropTypes from 'prop-types';
import { getCount, getPathToS3Bucket } from 'lib/utils';
import truncate from 'lodash/truncate';
import {
  Button, Card, Modal, Image, List, Loader, Popup
} from 'semantic-ui-react';
import iconVideoCamera from 'static/icons/icon_32px_videoCamera.png';
import iconRemove from 'static/icons/icon_remove.svg';
import iconReplace from 'static/icons/icon_replace.svg';
import FileUploadProgressBar from 'components/admin/projects/ProjectEdit/FileUploadProgressBar/FileUploadProgressBar';
import { UploadContext } from '../VideoEdit/VideoEdit';
import './ProjectUnitItem.scss';

const EditSingleProjectItem = dynamic( () => import( /* webpackChunkName: "editSingleProjectItem" */ '../EditSingleProjectItem/EditSingleProjectItem' ) );
const GeneralError = dynamic( () => import( /* webpackChunkName: "generalError" */ 'components/errors/GeneralError/GeneralError' ) );

const ProjectUnitItem = props => {
  const { projectId, unit, filesToUpload } = props;

  // const PLACEHOLDER = '/static/images/thumbnail_video.jpg';
  const PLACEHOLDER = null;

  const [thumbnail, setThumbnail] = useState( PLACEHOLDER );
  const [title, setTitle] = useState( '' );
  const [unitUploadComplete, setUnitUploadComplete] = useState( false );
  const [error, setError] = useState( false );

  // do not put in state as it will create a new object and will not track upload
  // need a better solution as this is prone to errors
  const unitFileToUpload = filesToUpload.filter( file => file.language === unit.language.id );

  const uploadInProgress = useContext( UploadContext );

  // implemment subscriptions to track thumbnail changes
  const getThumbnail = u => {
    if ( u && u.thumbnails && u.thumbnails[0] && u.thumbnails[0].image ) {
      return `${getPathToS3Bucket()}/${u.thumbnails[0].image.url}`;
    }
    // return PLACEHOLDER;
  };

  const getStreams = () => {
    const filesCount = ( unit.files && unit.files.length ) || 0;
    if ( filesCount > 0 ) {
      return unit.files.reduce( ( acc, file ) => (
        [...acc, ...file.stream]
      ), [] );
    }
    return [];
  };

  const getStreamUrls = ( site = 'vimeo' ) => {
    const streams = getStreams();
    return streams.reduce( ( acc, stream ) => {
      if ( stream.site === site && stream.url ) {
        acc.push( stream.url );
      }
      return acc;
    }, [] );
  };

  const handleOnComplete = () => {
    setUnitUploadComplete( true );
  };

  useEffect( () => {
    setThumbnail( getThumbnail( unit ) );
    setTitle( unit && unit.title ? unit.title : '[Title]' );

    // expected uploads should equal successful uploads
    const hasError = getCount( unit.files ) > getCount( getStreamUrls() );
    setError( hasError );
  }, [unit] );


  const renderProjectItem = () => (
    <Card className="project-unit-item">
      <div className="image-wrapper">
        { thumbnail
          ? ( <Image src={ thumbnail } fluid /> )
          : <div className="placeholder" />
        }
        { /* Has an upload been triggered and if so, are the files applicable to this unit complete? */ }
        { uploadInProgress && !unitUploadComplete && ( <Loader active size="small" /> ) }

        { /* Remove file overlay if above conditions are met */ }
        { !!unitFileToUpload.length && !unitUploadComplete && (
          <List className="file-list">
            { unitFileToUpload.map( file => <List.Item key={ file.id }>{ truncate( file.input.name, { length: 45 } ) }</List.Item> ) }
          </List>
        )
        }

        { !error
          && <img className="metaicon" src={ iconVideoCamera } alt="Video Icon" /> }
      </div>

      { /* Remove progress bar if above conditions are met */ }
      { uploadInProgress && !unitUploadComplete && !!unitFileToUpload.length && (
        <FileUploadProgressBar
          filesToUpload={ unitFileToUpload }
          label="Upload in progress"
          labelAlign="left"
          barSize="tiny"
          onComplete={ handleOnComplete }
          showPercent
        />
      ) }
      <Card.Content className={ error ? 'error' : '' }>
        <Card.Header>{ title }</Card.Header>
        <Card.Meta>
          <List>
            <List.Item>Language: { unit.language.displayName }</List.Item>
            <List.Item>Files: { getCount( getStreamUrls() ) }</List.Item>
            { error
              && (
                <List.Item>
                  <GeneralError msg="Uploading Error">
                    <Button.Group
                      basic
                      className="actions"
                      size="mini"
                      style={ { border: 'none' } }
                    >
                      <Popup
                        trigger={ (
                          <Button className="replace" style={ { marginRight: 0, padding: '0.25rem' } }>
                            <img src={ iconReplace } alt="replace icon" />
                          </Button>
                        ) }
                        content="Replace this video unit"
                        hideOnScroll
                        inverted
                        on={ ['hover', 'focus'] }
                        size="mini"
                      />
                      <Popup
                        trigger={ (
                          <Button className="delete" style={ { border: 'none', padding: '0.25rem' } }>
                            <img src={ iconRemove } alt="delete icon" />
                          </Button>
                        ) }
                        content="Delete this video unit"
                        hideOnScroll
                        inverted
                        on={ ['hover', 'focus'] }
                        size="mini"
                      />
                    </Button.Group>
                  </GeneralError>
                </List.Item>
              ) }
          </List>
        </Card.Meta>
      </Card.Content>
    </Card>
  );

  if ( projectId && !error ) {
    return (
      <Modal
        key={ unit.language.id }
        trigger={ renderProjectItem() }
        closeIcon
      >
        <Modal.Content>
          <EditSingleProjectItem
            projectId={ projectId }
            itemId={ unit.id }
          />
        </Modal.Content>
      </Modal>
    );
  }

  // this is a new project, deactivate modal
  return renderProjectItem();
};

ProjectUnitItem.propTypes = {
  unit: PropTypes.object,
  filesToUpload: PropTypes.array,
  projectId: PropTypes.string
};


export default ProjectUnitItem;
