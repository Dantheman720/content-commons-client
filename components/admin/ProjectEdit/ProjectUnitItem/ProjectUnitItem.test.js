import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ProjectUnitItem from './ProjectUnitItem';

jest.mock( 'next-server/dynamic', () => () => 'Dynamic' );

jest.mock( 'next-server/config', () => () => ( { publicRuntimeConfig: { REACT_APP_AWS_S3_PUBLISHER_BUCKET: 's3-bucket-url' } } ) );

jest.mock(
  '../EditSingleProjectItem/EditSingleProjectItem',
  () => function EditSingleProjectItem() { return ''; }
);

const props = {
  projectId: '123',
  filesToUpload: [
    {
      language: 'cjsq439dz005607560gwe7k3m',
      use: 'cjsw78q6p00lp07566znbyatd',
      quality: 'BROADCAST',
      videoBurnedInStatus: 'CLEAN',
      input: {
        name: 'Mexico City 1.mp4',
        lastModified: 1517162208000,
        size: 15829673,
        type: 'video/mp4'
      },
      id: 'af1557de-c2ad-4107-bcd3-369f05779320',
      loaded: 0
    },
    {
      language: 'cjsq439dz005607560gwe7k3m',
      use: 'cjsw78q6p00lp07566znbyatd',
      quality: 'WEB',
      videoBurnedInStatus: 'CLEAN',
      input: {
        name: 'Mexico City 2.mp4',
        lastModified: 1517162208000,
        size: 15829673,
        type: 'video/mp4'
      },
      id: 'bdef2cb3-169b-4d98-8aec-f99442e6dcfa',
      loaded: 0
    }
  ],
  unit: {
    id: 'cjzy5mcr41bf10720njs7wa7m',
    updatedAt: '2019-08-30T13:29:50.596Z',
    title: 'the project title',
    descPublic: null,
    files: [
      {
        id: 'cjzy5mcs01bf20720add2sklq',
        createdAt: '2019-08-30T13:29:50.596Z',
        updatedAt: '2019-08-30T13:29:50.596Z',
        filename: 'Mexico City 1.mp4',
        filetype: 'video/mp4',
        filesize: 15829673,
        duration: 14.889875,
        videoBurnedInStatus: 'SUBTITLED',
        quality: 'WEB',
        dimensions: {
          id: 'cjzy5mcsg1bf40720p9jxyixl',
          width: 1920,
          height: 1080,
          __typename: 'Dimensions'
        },
        url: '2019/08/commons.america.gov_cjzy5lpzv1bef07207s354cbw/mexico_city_1.mp4',
        language: {
          id: 'cjsq439dz005607560gwe7k3m',
          locale: 'en-us',
          languageCode: 'en',
          displayName: 'English',
          textDirection: 'LTR',
          nativeName: 'English',
          __typename: 'Language'
        },
        use: {
          id: 'cjsw78q6p00lp07566znbyatd',
          name: 'Full Video',
          __typename: 'VideoUse'
        },
        stream: [
          {
            id: 'cjzy5mcsj1bf50720wxk4bdsf',
            site: 'vimeo',
            url: 'https://vimeo.com/356893849',
            __typename: 'VideoStream'
          }
        ],
        __typename: 'VideoFile'
      }
    ],
    language: {
      id: 'cjsq439dz005607560gwe7k3m',
      locale: 'en-us',
      languageCode: 'en',
      displayName: 'English',
      textDirection: 'LTR',
      nativeName: 'English',
      __typename: 'Language'
    },
    thumbnails: [
      {
        id: 'cjzy5me4v1bfm0720854bc97o',
        size: 'FULL',
        image: {
          id: 'cjzy5mcsu1bf60720zyt05l8l',
          createdAt: '2019-08-30T13:29:50.596Z',
          updatedAt: '2019-08-30T13:29:50.596Z',
          filename: 'Mexico City 1.jpg',
          filetype: 'image/jpeg',
          filesize: 1030591,
          url: '2019/08/commons.america.gov_cjzy5lpzv1bef07207s354cbw/mexico_city_1.jpg',
          alt: null,
          use: {
            id: 'cjtkdq8kr0knf07569goo9eqe',
            name: 'Thumbnail/Cover Image',
            __typename: 'ImageUse'
          },
          language: {
            id: 'cjsq439dz005607560gwe7k3m',
            locale: 'en-us',
            languageCode: 'en',
            displayName: 'English',
            textDirection: 'LTR',
            nativeName: 'English',
            __typename: 'Language'
          },
          __typename: 'ImageFile'
        },
        __typename: 'Thumbnail'
      }
    ],
    tags: [],
    __typename: 'VideoUnit'
  },
};

const Component = <ProjectUnitItem { ...props } />;

describe( '<ProjectUnitItem />', () => {
  it( 'renders without crashing', () => {
    const wrapper = mount( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );
} );
