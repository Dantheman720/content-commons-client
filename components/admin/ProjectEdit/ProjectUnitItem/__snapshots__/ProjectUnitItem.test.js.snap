// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectUnitItem /> renders without crashing 1`] = `
<ProjectUnitItem
  filesToUpload={
    Array [
      Object {
        "id": "af1557de-c2ad-4107-bcd3-369f05779320",
        "input": Object {
          "lastModified": 1517162208000,
          "name": "Mexico City 1.mp4",
          "size": 15829673,
          "type": "video/mp4",
        },
        "language": "cjsq439dz005607560gwe7k3m",
        "loaded": 0,
        "quality": "BROADCAST",
        "use": "cjsw78q6p00lp07566znbyatd",
        "videoBurnedInStatus": "CLEAN",
      },
      Object {
        "id": "bdef2cb3-169b-4d98-8aec-f99442e6dcfa",
        "input": Object {
          "lastModified": 1517162208000,
          "name": "Mexico City 2.mp4",
          "size": 15829673,
          "type": "video/mp4",
        },
        "language": "cjsq439dz005607560gwe7k3m",
        "loaded": 0,
        "quality": "WEB",
        "use": "cjsw78q6p00lp07566znbyatd",
        "videoBurnedInStatus": "CLEAN",
      },
    ]
  }
  projectId="123"
  unit={
    Object {
      "__typename": "VideoUnit",
      "descPublic": null,
      "files": Array [
        Object {
          "__typename": "VideoFile",
          "createdAt": "2019-08-30T13:29:50.596Z",
          "dimensions": Object {
            "__typename": "Dimensions",
            "height": 1080,
            "id": "cjzy5mcsg1bf40720p9jxyixl",
            "width": 1920,
          },
          "duration": 14.889875,
          "filename": "Mexico City 1.mp4",
          "filesize": 15829673,
          "filetype": "video/mp4",
          "id": "cjzy5mcs01bf20720add2sklq",
          "language": Object {
            "__typename": "Language",
            "displayName": "English",
            "id": "cjsq439dz005607560gwe7k3m",
            "languageCode": "en",
            "locale": "en-us",
            "nativeName": "English",
            "textDirection": "LTR",
          },
          "quality": "WEB",
          "stream": Array [
            Object {
              "__typename": "VideoStream",
              "id": "cjzy5mcsj1bf50720wxk4bdsf",
              "site": "vimeo",
              "url": "https://vimeo.com/356893849",
            },
          ],
          "updatedAt": "2019-08-30T13:29:50.596Z",
          "url": "2019/08/commons.america.gov_cjzy5lpzv1bef07207s354cbw/mexico_city_1.mp4",
          "use": Object {
            "__typename": "VideoUse",
            "id": "cjsw78q6p00lp07566znbyatd",
            "name": "Full Video",
          },
          "videoBurnedInStatus": "SUBTITLED",
        },
      ],
      "id": "cjzy5mcr41bf10720njs7wa7m",
      "language": Object {
        "__typename": "Language",
        "displayName": "English",
        "id": "cjsq439dz005607560gwe7k3m",
        "languageCode": "en",
        "locale": "en-us",
        "nativeName": "English",
        "textDirection": "LTR",
      },
      "tags": Array [],
      "thumbnails": Array [
        Object {
          "__typename": "Thumbnail",
          "id": "cjzy5me4v1bfm0720854bc97o",
          "image": Object {
            "__typename": "ImageFile",
            "alt": null,
            "createdAt": "2019-08-30T13:29:50.596Z",
            "filename": "Mexico City 1.jpg",
            "filesize": 1030591,
            "filetype": "image/jpeg",
            "id": "cjzy5mcsu1bf60720zyt05l8l",
            "language": Object {
              "__typename": "Language",
              "displayName": "English",
              "id": "cjsq439dz005607560gwe7k3m",
              "languageCode": "en",
              "locale": "en-us",
              "nativeName": "English",
              "textDirection": "LTR",
            },
            "updatedAt": "2019-08-30T13:29:50.596Z",
            "url": "2019/08/commons.america.gov_cjzy5lpzv1bef07207s354cbw/mexico_city_1.jpg",
            "use": Object {
              "__typename": "ImageUse",
              "id": "cjtkdq8kr0knf07569goo9eqe",
              "name": "Thumbnail/Cover Image",
            },
          },
          "size": "FULL",
        },
      ],
      "title": "the project title",
      "updatedAt": "2019-08-30T13:29:50.596Z",
    }
  }
>
  <Modal
    centered={true}
    closeIcon={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    key="cjsq439dz005607560gwe7k3m"
    trigger={
      <Card
        className="project-unit-item"
      >
        <div
          className="image-wrapper"
        >
          <Image
            as="img"
            fluid={true}
            src="https://s3-bucket-url.s3.amazonaws.com/2019/08/commons.america.gov_cjzy5lpzv1bef07207s354cbw/mexico_city_1.jpg"
            ui={true}
          />
          <List
            className="file-list"
          >
            <ListItem>
              Mexico City 1.mp4
            </ListItem>
            <ListItem>
              Mexico City 2.mp4
            </ListItem>
          </List>
          <img
            alt="Video Icon"
            className="metaicon"
            src={Object {}}
          />
        </div>
        <CardContent
          className=""
        >
          <CardHeader>
            the project title
          </CardHeader>
          <CardMeta>
            <List>
              <ListItem>
                Language: 
                English
              </ListItem>
              <ListItem>
                Files: 
                1
              </ListItem>
            </List>
          </CardMeta>
        </CardContent>
      </Card>
    }
  >
    <Portal
      closeOnDocumentClick={false}
      closeOnEscape={true}
      eventPool="Modal"
      mountNode={<body />}
      onClose={[Function]}
      onMount={[Function]}
      onOpen={[Function]}
      onUnmount={[Function]}
      openOnTriggerClick={true}
      trigger={
        <Card
          className="project-unit-item"
        >
          <div
            className="image-wrapper"
          >
            <Image
              as="img"
              fluid={true}
              src="https://s3-bucket-url.s3.amazonaws.com/2019/08/commons.america.gov_cjzy5lpzv1bef07207s354cbw/mexico_city_1.jpg"
              ui={true}
            />
            <List
              className="file-list"
            >
              <ListItem>
                Mexico City 1.mp4
              </ListItem>
              <ListItem>
                Mexico City 2.mp4
              </ListItem>
            </List>
            <img
              alt="Video Icon"
              className="metaicon"
              src={Object {}}
            />
          </div>
          <CardContent
            className=""
          >
            <CardHeader>
              the project title
            </CardHeader>
            <CardMeta>
              <List>
                <ListItem>
                  Language: 
                  English
                </ListItem>
                <ListItem>
                  Files: 
                  1
                </ListItem>
              </List>
            </CardMeta>
          </CardContent>
        </Card>
      }
    >
      <Ref
        innerRef={[Function]}
      >
        <Card
          className="project-unit-item"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="ui card project-unit-item"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="image-wrapper"
            >
              <Image
                as="img"
                fluid={true}
                src="https://s3-bucket-url.s3.amazonaws.com/2019/08/commons.america.gov_cjzy5lpzv1bef07207s354cbw/mexico_city_1.jpg"
                ui={true}
              >
                <img
                  className="ui fluid image"
                  src="https://s3-bucket-url.s3.amazonaws.com/2019/08/commons.america.gov_cjzy5lpzv1bef07207s354cbw/mexico_city_1.jpg"
                />
              </Image>
              <List
                className="file-list"
              >
                <div
                  className="ui list file-list"
                  role="list"
                >
                  <ListItem
                    key="af1557de-c2ad-4107-bcd3-369f05779320"
                  >
                    <div
                      className="item"
                      onClick={[Function]}
                      role="listitem"
                    >
                      Mexico City 1.mp4
                    </div>
                  </ListItem>
                  <ListItem
                    key="bdef2cb3-169b-4d98-8aec-f99442e6dcfa"
                  >
                    <div
                      className="item"
                      onClick={[Function]}
                      role="listitem"
                    >
                      Mexico City 2.mp4
                    </div>
                  </ListItem>
                </div>
              </List>
              <img
                alt="Video Icon"
                className="metaicon"
                src={Object {}}
              />
            </div>
            <CardContent
              className=""
            >
              <div
                className="content"
              >
                <CardHeader>
                  <div
                    className="header"
                  >
                    the project title
                  </div>
                </CardHeader>
                <CardMeta>
                  <div
                    className="meta"
                  >
                    <List>
                      <div
                        className="ui list"
                        role="list"
                      >
                        <ListItem>
                          <div
                            className="item"
                            onClick={[Function]}
                            role="listitem"
                          >
                            Language: 
                            English
                          </div>
                        </ListItem>
                        <ListItem>
                          <div
                            className="item"
                            onClick={[Function]}
                            role="listitem"
                          >
                            Files: 
                            1
                          </div>
                        </ListItem>
                      </div>
                    </List>
                  </div>
                </CardMeta>
              </div>
            </CardContent>
          </a>
        </Card>
      </Ref>
    </Portal>
  </Modal>
</ProjectUnitItem>
`;
